OrbitClientGgp/ClientGgp.cpp

```diff
@@ -79,9 +78,107 @@ bool ClientGgp::StopCapture() {
+  process->SetIs64Bit(process_it->is_64_bit());
+  LOG("Process info: pid:%d, name:%s, path:%s, is64:%d", process->GetId(), process->GetName(),
+      process->GetFullPath(), process->GetIs64Bit());
+  return process;
+}
+
+bool ClientGgp::LoadModuleAndSymbols() {
+  // Load modules for target_process_
+  ErrorMessageOr<std::vector<ModuleInfo>> result_module_infos =
+      process_client_->LoadModuleList(target_process_->GetId());
+  if (result_module_infos.has_error()) {
+    ERROR("Error retrieving modules: %s", result_module_infos.error().message());
+    return false;
+  }
+  // Find module that corresponds to the binary of target_process
+  const std::vector<ModuleInfo>& module_infos = result_module_infos.value();
+  LOG("List of modules");
+  for (const ModuleInfo& info : module_infos) {
+    LOG("name:%s, path:%s, size:%d, address_start:%d. address_end:%d, build_id:%s", info.name(),
+        info.file_path(), info.file_size(), info.address_start(), info.address_end(),
+        info.build_id());
+  }
+  std::string_view main_executable_name = target_process_->GetName();
+  auto module_it = find_if(module_infos.begin(), module_infos.end(),
+                           [&main_executable_name](const ModuleInfo& info) {
+                             return info.name() == main_executable_name;
+                           });
+  if (module_it == module_infos.end()) {
+    ERROR("Error: Module correspondent to process binary not found");
+    return false;
+  }
+  LOG("Found module correspondent to process binary");
+  std::shared_ptr<Module> module = std::make_shared<Module>();
+  module->m_Name = module_it->name();
+  module->m_FullName = module_it->file_path();
+  module->m_PdbSize = module_it->file_size();
+  module->m_AddressStart = module_it->address_start();
+  module->m_AddressEnd = module_it->address_end();
+  module->m_DebugSignature = module_it->build_id();
+  target_process_->AddModule(module);
+  LOG("Module info: name:%s, path:%s, size:%d, address_start:%d. address_end:%d, build_id:%s",
+      module->m_Name, module->m_FullName, module->m_PdbSize, module->m_AddressStart,
+      module->m_AddressEnd, module->m_DebugSignature);
+
+  // Load symbols for the module
+  const std::string& module_path = module->m_FullName;
+  LOG("Looking for debug info file for %s", module_path);
+  ErrorMessageOr<std::string> result_main_executable_debug_file =
+      process_client_->FindDebugInfoFile(module_path);
+  if (result_main_executable_debug_file.has_error()) {
+    ERROR("Error loading symbols: %s", result_main_executable_debug_file.error().message());
+    return false;
+  }
+  const std::string& main_executable_debug_file = result_main_executable_debug_file.value();
+  LOG("Found file: %s", main_executable_debug_file);
+  LOG("Loading symbols");
+  ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> result_symbols =
+      symbol_helper_.LoadSymbolsFromFile(main_executable_debug_file, module->m_DebugSignature);
```

Comment:
Do I understand this correctly, this code (ClientGgp) is also executed on the instance? This loading will not work when this is run on a workstation.

Reply:
Yes exactly, ClientGgp runs on the instance